pkgs/development/python-modules/mocket/default.nix

```diff
@@ -1,62 +1,72 @@
 { lib
+, aiohttp
+, asgiref
 , buildPythonPackage
-, fetchPypi
-, pythonOlder
-, isPy3k
 , decorator
-, http-parser
-, python-magic
-, urllib3
-, pytestCheckHook
-, pytest-mock
-, aiohttp
 , fastapi
+, fetchPypi
 , gevent
+, httptools
 , httpx
+, isPy3k
+, pook
+, pytest-mock
+, pytestCheckHook
+, python-magic
+, pythonOlder
 , redis
 , requests
 , sure
-, pook
+, urllib3
 }:
 
 buildPythonPackage rec {
   pname = "mocket";
-  version = "3.10.9";
-  disabled = !isPy3k;
+  version = "3.11.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "sha256-fAVw5WvpJOITQWqA8Y6Xi7QbaunZ1WGXxAuUMXbh+Aw=";
+    hash = "sha256-OIdLP3hHnPZ9MqrHt6G5t2SSO342+jTACgzxM6RjVYM=";
   };
 
   propagatedBuildInputs = [
     decorator
-    http-parser
+    httptools
     python-magic
     urllib3
   ];
 
+  passthru.optional-dependencies = {
+    pook = [
+      pook
+    ];
+  };
+
   nativeCheckInputs = [
-    pytestCheckHook
-    pytest-mock
     aiohttp
+    asgiref
     fastapi
     gevent
     httpx
+    pytest-mock
+    pytestCheckHook
     redis
     requests
     sure
-    pook
-  ];
+  ] ++ passthru.optional-dependencies.pook;
 
-  # skip http tests
+  # Skip http tests
   SKIP_TRUE_HTTP = true;
-  pytestFlagsArray = [
+
+  disabledTestPaths = [
     # Requires a live Redis instance
-    "--ignore=tests/main/test_redis.py"
+    "tests/main/test_redis.py"
   ] ++ lib.optionals (pythonOlder "3.8") [
```

Comment:
Can be removed. We don't support such old versions anymore.

Reply:
Thanks, removed.